nomad/state/state_store.go

```diff
@@ -279,9 +280,11 @@ func (s *StateStore) SnapshotMinIndex(ctx context.Context, index uint64) (*State
 			retryTimer = time.NewTimer(backoffBase)
 		} else {
 			// Subsequent retry, reset timer
-			deadline := 1 << (2 * retries) * backoffBase
-			if deadline > backoffLimit {
-				deadline = backoffLimit
+			if deadline < backoffLimit {
+				deadline = 1 << (2 * retries) * backoffBase
+				if deadline > backoffLimit {
+					deadline = backoffLimit
+				}
```

Comment:
care to put `retries++` from line 276 down here too?  I think the end result is the same with the new extra `deadline < limit` check guarding the `2 * retries`, but I like how the others have their `retries`-equivalent var right after where it's used.  even if it's not used in the math, I think this one would continue incrementing the var arbitrarily-long which just tastes funny to me.

Reply:
Makes sense.